polars/polars-ops/src/chunked_array/strings/namespace.rs

```diff
@@ -395,6 +397,13 @@ pub trait Utf8NameSpaceImpl: AsUtf8 {
         ca.apply(|s| str::to_uppercase(s).into())
     }
 
+    #[must_use]
+    #[cfg(feature = "string_inflection")]
+    fn to_title_case(&self) -> Utf8Chunked {
+        let ca = self.as_utf8();
+        ca.apply(|s| cruet::to_title_case(s).into())
```

Comment:
This will allocate a `String` on every iteration.

We can inline the `cruet` function (this also saves a dependency) and amortize the allocation  string outside the loop.

This is the function we should inline. 

https://github.com/chrislearn/cruet/blob/c7fe28f04693912cfcaa564bcec4fe0ef8ef5d79/src/case/mod.rs#L105

The amortization can take inspiration from `str().replace` which uses `apply_mut`.

Reply:
Thanks for the pointer - indeed I'll vendor that code.

Figured I'd first connect the dots and then properly implement it.

Reply:
I was also considering maybe implementing some of the other inflections, but none seem too appealing at this point.

Reply:
str.to_capitalizecase ?

Reply:
> Figured I'd first connect the dots and then properly implement it.

Right, the dots look good. :+1: